great_ape_safe/ape_api/curve.py

```diff
@@ -163,10 +165,20 @@ def withdraw(self, lp_token, mantissa):
         # (in same ratio as pool is currently in)
         # https://curve.readthedocs.io/exchange-pools.html#StableSwap.remove_liquidity
         pool = self._get_pool_from_lp_token(lp_token)
-
-        n_coins = self._get_n_coins(lp_token)
-        # TODO: slippage and stuff
-        minima = list(np.zeros(n_coins))
+        coins = self._get_coins(lp_token)
+
+        pool_usd_values = [
+            get_cg_price(x)
+            * (pool.balances(i) / 10 ** self.safe.contract(x).decimals())
+            for i, x in enumerate(coins)
+        ]
```

Comment:
ideally here we would use `pool.price_oracle` instead of coingecko. the problem is for some functions, i couldnt find a clear mapping from coin address to its index. for example, for [tricrypto2 pool](https://etherscan.io/address/0xa895B89D74a6BC23a284a0526e123ea776674cF5#readContract) index `0`:
- `pool.price_oracle(0)` -> wbtc price
- `pool.coins(0)` -> usdt address

is there an intuitive way to do this programmatically?

Reply:
one think to consider is that not all pools has as a feature the internal oracle, e.g: [bveCVX_CVX_f](https://etherscan.io/address/0x04c90C198b2eFF55716079bc06d7CCc4aa4d7512#readContract) 

what are your thoughts on doing the dry-run of the method itself with [call](https://eth-brownie.readthedocs.io/en/stable/api-network.html#ContractTx.call) and use the returned amounts as the minima (those amounts given back can be seen as non-manipulated) and in the particular cases where curve method may not return anything, which i recall there was some cases fallback to the cg/oracle approach?

Reply:
great idea, that should work. will make it foolproof too

Reply:
69fbf12